### ISSUE TITLE
Localization of time broken in the set editor

### ISSUE THREAD

**Issue by robert-marik** [openwebwork/webwork2#1164]
WeBWorK allows time localization, but the set editor assumes time in English format when parsing start time, due time etc.

Steps to reproduce the problem:

1. set  `$siteDefaults{locale}="cs_CZ";`  in site.conf file, restart apache.
2. create hw set which starts in afternoon, say at 3:00pm
3. if you open the set again (to edit problem, for example), the start time is `...(date).. at 03:00odp.`, where "odp." is localization for "pm"
4. after saving the changes, WeBWorK assumes English date format, see the file Utils.pm where you can find
```
sub unformatDateAndTime {
        my ($string) = @_;
        my $orgString = $string;

        $string =~ s|^\s+||;
        $string =~ s|\s+$||;
        $string =~ s|at| at |i; ## OK if forget to enter spaces or use wrong case
        $string =~ s|AM| AM|i;  ## OK if forget to enter spaces or use wrong case
        $string =~ s|PM| PM|i;  ## OK if forget to enter spaces or use wrong case
        $string =~ s|,| at |;   ## start translating old form of date/time to new form

        # case where the at is missing: MM/DD/YYYY at HH:MM AMPM ZONE
        unformatDateAndTime_error($orgString, "The 'at' appears to be missing.")
                if $string =~ m|^\s*[\/\d]+\s+[:\d]+|;

        my ($date, $at, $time, $AMPM, $TZ) = split /\s+/, $string;
```
and then (few lines below)
```
	# default value for $AMPM
	$AMPM = "AM" unless defined $AMPM;
```
whis is followed by 
```
	$pm = 0;
	$pm = 12 if ($AMPM =~/PM/ and $hour < 12);
	$hour += $pm;
	$hour = 0 if ($AMPM =~/AM/ and $hour == 12);
```
5. as a consequence of bad parsing, the time is silently changed from 3:00pm to 3:00am

In other words, the AM and PM strings are hardwired to WeBWorK. Could be resolved in https://github.com/openwebwork/webwork2/pull/1333

**Comment by robert-marik** [comment]
Fixed in #1615

**Comment by robert-marik** [comment]
closing
